code/modules/research/xenobiology/vatgrowing/microscope.dm

```diff
@@ -5,28 +5,53 @@
 	icon_state = "microscope"
 	var/obj/item/petri_dish/current_dish
 
-/obj/structure/microscope/attacked_by(obj/item/I, mob/living/user)
-	if(!istype(I, /obj/item/petri_dish))
-		return ..()
-	if(current_dish)
-		to_chat(user, span_warning("There is already a petridish in \the [src]."))
-		return
-	to_chat(user, span_notice("You put [I] into \the [src]."))
-	current_dish = I
-	current_dish.forceMove(src)
+/obj/structure/microscope/Initialize(mapload)
+	. = ..()
+	var/static/list/hovering_item_typechecks = list(
+		/obj/item/petri_dish = list(
+			SCREENTIP_CONTEXT_LMB = "Add petri dish",
+		),
+	)
+	AddElement(/datum/element/contextual_screentip_item_typechecks, hovering_item_typechecks)
+	AddElement(/datum/element/contextual_screentip_bare_hands, rmb_text = "Remove petri dish")
+
+/obj/structure/microscope/item_interaction(mob/living/user, obj/item/tool, list/modifiers)
+	if(istype(tool, /obj/item/petri_dish))
```

Comment:
pls early return this

Reply:
made the code below into a separate proc instead